packages/x-date-pickers/src/internals/hooks/usePickerState.ts

```diff
@@ -278,7 +294,16 @@ export const usePickerState = <TInputValue, TValue, TDate>(
         setDate({ value: valueManager.getTodayValue(utils), action: 'acceptAndClose' });
       },
     }),
-    [setDate, isOpen, utils, dateState, valueManager],
+    [
+      isOpen,
+      setDate,
+      valueManager,
+      value,
+      dateState.draft,
+      dateState.committed,
+      dateState.resetFallback,
+      utils,
+    ],
```

Comment:
There should be difference, especially in this case.
I've changed it to this style only because it was the change that the IDE has done as auto suggestion.
WiIl add `value` manually for cleanliness.